tests/conftest.py

```diff
@@ -2910,3 +2910,77 @@ def finalizer():
+    ready_count = 0
+    for ep_info in pods_info:
+        if ep_info['status']['containerStatuses'][0]['ready']:
+            ready_count += 1
+    return ready_count
+
+
+@pytest.fixture(scope="function")
+def nb_ensure_endpoint_count(request):
+    """
+    Validate and ensure the number of running noobaa endpoints
+    """
+    cls = request.cls
+    min_ep_count = cls.MIN_ENDPOINT_COUNT
+    max_ep_count = cls.MAX_ENDPOINT_COUNT
+
+    assert min_ep_count <= max_ep_count
+    namespace = defaults.ROOK_CLUSTER_NAMESPACE
+    should_wait = False
+
+    # prior to 4.6 we configured the ep count directly on the noobaa cr.
+    if float(config.ENV_DATA['ocs_version']) < 4.6:
+        noobaa = OCP(kind='noobaa', namespace=namespace)
+        resource = noobaa.get()['items'][0]
+        resource_name = resource['metadata']['name']
+        endpoints = resource.get('spec', {}).get('endpoints', {})
+
+        if endpoints.get('minCount', -1) != min_ep_count:
+            log.info(f"Changing minimum Noobaa endpoints to {min_ep_count}")
+            params = f'{{"spec":{{"endpoints":{{"minCount":{min_ep_count}}}}}}}'
+            noobaa.patch(resource_name='noobaa', params=params, format_type='merge')
+            should_wait = True
+
+        if endpoints.get('maxCount', -1) != max_ep_count:
+            log.info(f"Changing minimum Noobaa endpoints to {max_ep_count}")
+            params = f'{{"spec":{{"endpoints":{{"minCount":{max_ep_count}}}}}}}'
+            noobaa.patch(resource_name='noobaa', params=params, format_type='merge')
+            should_wait = True
+
+    else:
+        storage_cluster = OCP(kind=constants.STORAGECLUSTER, namespace=namespace)
+        resource = storage_cluster.get()['items'][0]
+        resource_name = resource['metadata']['name']
+        endpoints = resource.get('spec', {}).get('multiCloudGateway', {}).get('endpoints', {})
+
+        if endpoints.get('minCount', -1) != min_ep_count:
+            log.info(f"Changing minimum Noobaa endpoints to {min_ep_count}")
+            params = f'{{"spec":{{"multiCloudGateway":{{"endpoints":{{"minCount":{min_ep_count}}}}}}}}}'
+            storage_cluster.patch(resource_name=resource_name, params=params, format_type='merge')
+            should_wait = True
+
+        if endpoints.get('maxCount', -1) != max_ep_count:
+            log.info(f"Changing maximum Noobaa endpoints to {max_ep_count}")
+            params = f'{{"spec":{{"multiCloudGateway":{{"endpoints":{{"maxCount":{max_ep_count}}}}}}}}}'
+            storage_cluster.patch(resource_name=resource_name, params=params, format_type='merge')
+            should_wait = True
+
+    if should_wait:
+        # loggin the ready endpoint count for 5 min in 30sec internvals
+        for i in range(10):
+            ready_count = get_ready_noobaa_endpoint_count(namespace)
+            log.info(f"[{i * 30}]: Waiting for noobaa endpoints to stabilize, current ready count: {ready_count}")
```

Comment:
what is `[{i * 30}]`?

Reply:
I am just printing the number of seconds passed since the start of the check for tracing purposes. 
I find it very helpful while diagnosing logs, it makes it easy to get a sense of how much time we really need if we decide to fine-tune the timeout in the future.

Reply:
ok. Maybe add something like `Time passed - {i * 30}...`

Reply:
The linter is limiting me to 120 chars. I already minimized most of the sentence

Reply:
You can do:

```python
log.info(
    f"Time passed - {i * 30} waiting for noobaa endpoints to stabilize, current ready "
    f"count: {ready_count}"
)
```

Reply:
Done, thanks